### ISSUE TITLE
cmd/link: CL 20968 breaks go install -buildmode=shared -linkshared std

### ISSUE THREAD

**Issue by paulzhol** [golang/go#14963]
1. What version of Go are you using (`go version`)?
`go version devel +ea0386f Fri Mar 25 11:00:48 2016 +0000 linux/amd64`

2. What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH=""
GORACE=""
GOROOT="/home/yuval/go.git"
GOTOOLDIR="/home/yuval/go.git/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build199985141=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

3. What did you do?
```
cd src
./make.bash
../bin/go install -buildmode=shared -linkshared std
```

4. What did you expect to see?
success

5. What did you see instead?
```
---
Installed Go for linux/amd64 in /home/yuval/go.git
Installed commands in /home/yuval/go.git/bin

../bin/go install -buildmode=shared -linkshared std
# /tmp/go-build305205369/libstd.so
inconsistent types for go.string."archive/tar: unknown file mode %v" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."/" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."tar: FileInfo is nil" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."<nil>" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."path" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."linkpath" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.major" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.minor" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.name" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.map" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.size" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."GNU.sparse.realsize" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."0.1" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."1.0" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."0.0" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."." and its Outer go.string.* (5 != 13)
inconsistent types for go.string."SCHILY.xattr." and its Outer go.string.* (5 != 13)
inconsistent types for go.string."uid" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."gid" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."atime" and its Outer go.string.* (5 != 13)
inconsistent types for go.string."size" and its Outer go.string.* (5 != 13)
/home/yuval/go.git/pkg/tool/linux_amd64/link: too many errors
```

I'm not sure if they are related, but the android builders also fail with similar erros:
http://build.golang.org/log/712f62b08da37026c4074e255b80b8c791069f57
http://build.golang.org/log/db84461adaf754e04b04d073efb9d530e09b7673
http://build.golang.org/log/412a22e79eba102ff13f99ebb6507be8ed626375

**Comment by ianlancetaylor** [comment]
Some strings are SGOSTRING and some are SGOSTRINGRELRO.

**Comment by crawshaw** [comment]
Fixed in 4d92041 (https://golang.org/cl/21110).
